mmdet/apis/ote/apis/detection/inference_task.py

```diff
@@ -168,29 +171,62 @@ def _create_model(config: Config, from_scratch: bool = False):
-                        Rectangle(x1=coords[0], y1=coords[1], x2=coords[2], y2=coords[3]),
-                        labels=assigned_label))
+            if self._task_type == TaskType.DETECTION and isinstance(all_results, list):
+                for label_idx, detections in enumerate(all_results):
+                    for i in range(detections.shape[0]):
+                        probability = float(detections[i, 4])
+                        coords = detections[i, :4].astype(float).copy()
+                        coords /= np.array([width, height, width, height], dtype=float)
+                        coords = np.clip(coords, 0, 1)
+
+                        if probability < confidence_threshold:
+                            continue
+
+                        assigned_label = [ScoredLabel(self._labels[label_idx],
+                                                      probability=probability)]
+                        if coords[3] - coords[1] <= 0 or coords[2] - coords[0] <= 0:
+                            continue
+
+                        shapes.append(Annotation(
+                            Rectangle(x1=coords[0], y1=coords[1], x2=coords[2], y2=coords[3]),
+                            labels=assigned_label))
+            elif self._task_type == TaskType.COUNTING and isinstance(all_results, tuple):
+                box_results, mask_results = all_results
+                for label_idx, masks in enumerate(mask_results):
+                    probs = box_results[label_idx][:, 4]
+                    for i, mask in enumerate(masks):
+                        mask = mask.astype(np.uint8)
+                        contours, hierarchies = cv2.findContours(
+                          mask, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE)
+                        if hierarchies is None:
+                            continue
+                        for contour, hierarchy in zip(contours, hierarchies[0]):
+                            if hierarchy[3] != -1:
+                                continue
+                            # TODO(ikrylov): MinAreaRect box points should be returned in case of Rotated Object Detection
+                            # rect = cv2.minAreaRect(contour)
+                            # box = cv2.boxPoints(rect)
+                            contour = list(contour)
+                            if len(contour) <= 2 or probs[i] < confidence_threshold:
+                                continue
+                            points = [
+                                Point(
+                                  x=point[0][0] / width,
+                                  y=point[0][1] / height) for point in contour]
+                            shapes.append(Annotation(
+                              Polygon(points=points),
+                              labels=[
+                                ScoredLabel(self._labels[label_idx],
+                                probability=probs[i])],
+                              id=label_idx,))
```

Comment:
```suggestion
                for label_idx, (boxes, masks) in enumerate(zip(*all_results)):
                    for i, (mask, probability) in enumerate(zip(masks, boxes[:, 4])):
                        mask = mask.astype(np.uint8)
                        contours, hierarchies = cv2.findContours(mask, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE)
                        if hierarchies is None:
                            continue
                        for contour, hierarchy in zip(contours, hierarchies[0]):
                            if hierarchy[3] != -1:
                                continue
                            # TODO(ikrylov): MinAreaRect box points should be returned in case of Rotated Object Detection
                            # rect = cv2.minAreaRect(contour)
                            # box = cv2.boxPoints(rect)
                            contour = list(contour)
                            if len(contour) <= 2 or probability < confidence_threshold:
                                continue
                            points = [Point(x=point[0][0] / width, y=point[0][1] / height) for point in contour]
                            labels = [ScoredLabel(self._labels[label_idx], probability=probability)]
                            shapes.append(Annotation(Polygon(points=points), labels=labels, id=label_idx))
```